src/Components/Components/src/BindMethods.cs

```diff
@@ -74,129 +74,187 @@ public static MulticastDelegate GetEventHandlerValue<T>(Func<T, Task> value)
         /// </summary>
         public static Action<UIEventArgs> SetValueHandler(Action<string> setter, string existingValue)
         {
-            return _ => setter((string)((UIChangeEventArgs)_).Value);
+            return _ =>
+            {
+                setter((string)((UIChangeEventArgs)_).Value);
+                ComponentBase.DispatchStateChangeAsync(setter.Target, Task.CompletedTask);
```

Comment:
Unless I'm missing something, it looks like this will still suffer from the same problem that killed the approach attempted in https://github.com/aspnet/Blazor/pull/1651, i.e., that `delegate.Target` won't be what we need if it's a capturing lambda.

Is that something you already have on the list to address for this PR?

The plan we previously came up with to fix this was https://github.com/aspnet/Blazor/pull/1651#issuecomment-438036193. It involves changing the codegen for events and bind.

Reply:
OK, I see you're working on addressing this in the commit titled `add event handler infrastructure`.

Reply:
Right. This isn't the fix for that, but these APIs need to change to preserve the existing semantics. These APIs needs to stay put because we still call them in the editor. It's the simplest thing to do to try and keep these APIs working until we completely transition.